File: romanToInt.cpp
#include <map>

class Solution {
public:
    int romanToInt(string s) {
        std::map<char, int> v;
        v['I'] = 1;
        v['V'] = 5;
        v['X'] = 10;
        v['L'] = 50;
        v['C'] = 100;
        v['D'] = 500;
        v['M'] = 1000;
        int sum = 0;
        for (int i = 0; i < s.length(); i++)
        {
            if (i+1 < s.length() && v[s[i]] < v[s[i+1]])
                sum -= v[s[i]];
            else
                sum += v[s[i]];
        }
        return sum;
    }
};

File: romanToInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        v = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        sum = 0;
        for i, x in enumerate(s):
            if i+1 < len(s) and v[x] < v[s[i+1]]:
                sum -= v[x];
            else:
                sum += v[x];
        return sum;
